drift_dev/lib/src/analyzer/dart/view_parser.dart

```diff
@@ -0,0 +1,237 @@
+      generateInsertable =
+          useRowClass.getField('generateInsertable')!.toBoolValue()!;
+
+      if (type is InterfaceType) {
+        existingClass = FoundDartClass(type.element, type.typeArguments);
+        name = type.element.name;
+      } else {
+        base.step.reportError(ErrorInDartCode(
+          message: 'The @UseRowClass annotation must be used with a class',
+          affectedElement: element,
+        ));
+      }
+    }
+
+    final verified = existingClass == null
+        ? null
+        : validateExistingClass(columns, existingClass,
+            constructorInExistingClass!, generateInsertable!, base.step.errors);
+    return _DataClassInformation(name, verified);
+  }
+
+  Future<String> _parseViewName(ClassElement element) async {
+    for (final annotation in element.metadata) {
+      final computed = annotation.computeConstantValue();
+      final annotationClass = computed!.type!.element!.name;
+
+      if (annotationClass == 'DriftView') {
+        final name = computed.getField('name')?.toStringValue();
+        if (name != null) {
+          return name;
+        }
+        break;
+      }
+    }
+
+    return element.name.snakeCase;
+  }
+
+  Future<Iterable<MoorColumn>> _parseColumns(ClassElement element) async {
+    final columnNames = element.allSupertypes
+        .map((t) => t.element)
+        .followedBy([element])
+        .expand((e) => e.fields)
+        .where((field) =>
+            isColumn(field.type) &&
+            field.getter != null &&
+            !field.getter!.isSynthetic)
+        .map((field) => field.name)
+        .toSet();
+
+    final fields = columnNames.map((name) {
+      final getter = element.getGetter(name) ??
+          element.lookUpInheritedConcreteGetter(name, element.library);
+      return getter!.variable;
+    });
+
+    final results = await Future.wait(fields.map((field) async {
+      final node =
+          await base.loadElementDeclaration(field.getter!) as MethodDeclaration;
+
+      return await base.parseColumn(node, field.getter!);
+    }));
+
+    return results.whereType();
+  }
+
+  Future<List<String>> _parseStaticReferences(
+      ClassElement element, List<MoorTable> tables) async {
+    return await Stream.fromIterable(element.allSupertypes
+            .map((t) => t.element)
+            .followedBy([element]).expand((e) => e.fields))
+        .asyncMap((field) => _getStaticReference(field, tables))
+        .where((ref) => ref != null)
+        .cast<String>()
+        .toList();
+  }
+
+  Future<String?> _getStaticReference(
+      FieldElement field, List<MoorTable> tables) async {
+    if (field.getter != null) {
+      try {
+        final node = await base.loadElementDeclaration(field.getter!);
+        if (node is MethodDeclaration && node.body is EmptyFunctionBody) {
+          final type = tables.firstWhereOrNull(
+              (tbl) => tbl.entityInfoName == node.returnType.toString());
```

Comment:
I think it would be awesome if users didn't have to use the name of the generated class. We should definitely support that still (for tables defined in drift files, for instance), but it would be great if they could use the Dart type too:

```dart
abstract class TodoCategoryItemCount extends View {
  TodoItems get todoItems;
  TodoCategories get todoCategories;
```

You can check if the table was declared in Dart and then use `tbl.fromClass` to check if that matches.

Reply:
That was my first thought, but you can't use this reference type. `.join([innerJoin(todoItems,..)])` in `as()` shows error because it isn't a `ResultSetImplemetation`. May `innerJoin()` can be modified to accept `Table`

Reply:
Oh, you're right. I think 47f0465795080944c48d135155bba05cafb67cf3 is not a good idea either (it'll break when joining views, for instance).

We could let users do something like this:

```dart
abstract class TodoCategoryItemCount extends View {
  @TodoItems
  TableInfo get todoItems;
```

But I'm not sure if that's more ergonomic. I'll keep pondering.

Reply:
I think this would be too boilerplate. Join parameter can be 'HasResultSet' that accepts views too, and not breaks current functionality.

Reply:
Other solution is separate view declaration from query and dsl view don't create a database level view. That was my first idea. Views are projections actually, queries isn't declared in DslView, have to be write like table queries. There is no performance gain of store view in database. Also, not migration friendly.